hash_practice/exercises.py

```diff
@@ -2,18 +2,50 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(nklog(k)), where k is the length of the string 
+        Space Complexity: O(log(n))
     """
-    pass
+    if not strings:
+        return []
+
+    result = {}
+    for word in strings:
+        key = ''.join(sorted(word))
+        if key in result:
+            result.get(key).append(word)
+        else:
+            result[key] = [word]
+    return result.values()
+    
+
+ 
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    if not nums:
+        return []
+    count = {}
+    frequency = [[] for i in range(len(nums)+1)]

```

Comment:
Small style nitpick: I usually prefer to wait to initialize a variable right before it is used, as that reduces the need to scroll back up. So I would put line 33 here down right before the `for` loop on line 39.